### PR TITLE
Improve DNS server handling in DHCP handlers

### PR DESCRIPTION
**Pull request by cschramm**
* Set router address as DNS server instead of loopback addresses
* Enable dnsmasq DNS service if possible and add DNS servers otherwise

Based on and includes #1564

### CODE REVIEW

**Comment by infirit** [comment]
I was planning on doing something similar so everything could be done through NM. So if this works for you merge it and I'll promise to test it :-)

File: blueman/main/NetConf.py [hunk 1]
```diff
@@ -166,318 +168,193 @@ def _read_dhcp_config() -> Tuple[str, str]:
...
-            error = p.communicate()[1]
-
-            # udhcpd takes time to create pid file
-            sleep(0.1)
+                                      "end": ipiface.network[-2],
+                                      "pid_path": self._pid_path}
 
-            pid = read_pid_file("/var/run/udhcpd.pan1.pid")
+    def _start(self, binary: str, ip4_address: str, ip4_mask: str, dns_servers: List[str]) -> Optional[bytes]:
+        config_file, self._config_path = mkstemp(prefix="udhcpd-")
```
Comment (infirit) [inline blueman/main/NetConf.py:228]: Would something like this work?

```
with os.fdopen(config_file, encoding="utf8") as f:
    f.write(self._generate_config(ip4_address, ip4_mask)
```

**Review by infirit — COMMENTED** [review]
Not tested but looks good in general. Let me know if you like me to actually test it out.

File: blueman/main/DNSServerProvider.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+from gi.repository import GObject
+
+from blueman.bluemantyping import GSignals
+
+
+class DNSServerProvider(GObject.GObject):
+    RESOLVER_PATH = "/etc/resolv.conf"
+
+    __gsignals__: GSignals = {"changed": (GObject.SignalFlags.NO_HOOKS, None, ())}
```
[the remark this answers is not in the document]
Comment (cschramm) [reply blueman/main/DNSServerProvider.py]: Right, that should've been part of #1553 but made it into #1564, the common base. Fixed.

**Review by infirit — APPROVED** [review]
Looks ok to me I'll test tonight. Your daily builds are still up right cause that would make it a lot easier to test in the VM.

**Comment by cschramm** [comment]
https://blueman.cschramm.eu/debian/ looks good. You'll need the python3.9 packages for bullseye.

**Comment by infirit** [comment]
Yup, everything works with your latest  git build.

**Pull request merged**